UnstoppableWallet/UnstoppableWallet/Core/Adapters/ZcashAdapter.swift

```diff
@@ -421,16 +460,24 @@ extension ZcashAdapter: ISendZcashAdapter {
     }
 
     func validate(address: String) throws {
-        guard !synchronizer.initializer.isValidTransparentAddress(address) else {
-            throw AppError.zcash(reason: .transparentAddress)
-        }
-
-        guard synchronizer.initializer.isValidShieldedAddress(address) else {
+        
+        guard address != receiveAddress else {
             throw AppError.addressInvalid
         }
+        
+        do {
+            let derivationTool = DerivationTool.default
+            
+            let validZAddress = try derivationTool.isValidShieldedAddress(address)
+            let validTAddress = try derivationTool.isValidTransparentAddress(address)
+            
+            guard validZAddress || validTAddress else {
+                throw AppError.addressInvalid
+            }
 
-        guard address != receiveAddress else {
-            throw AppError.zcash(reason: .sendToSelf)
+        } catch {
+            //FIXME: Should this be handled another way? logged? how?
```

Comment:
@ant013 I'd like to discuss how to handle this kind of interface differences on throwability or nullability